text_handler.py

```diff
@@ -30,37 +31,45 @@ async def __handleYoutubeDownload(self, message: types.Message):
             author = yt.author
             length = yt.length
             title = yt.title
+            escaped_title = re.escape(title)
             stream = (yt.streams
                       .filter(only_audio=True, file_extension='mp4')
                       .order_by('bitrate').desc().first())
             filesize = stream.filesize
-            video_url = "youtu.be/" + extract.video_id(message.text)
+            video_id = extract.video_id(message.text)
+            video_url = "youtu.be/" + video_id
             telegram_audio_limit = 52428800
 
             if (filesize < telegram_audio_limit and length < 600):
-                output_audiofile = stream.download("downloads")
-                output_thumbnail = "downloads/square_thumbnail.jpg"
+                output_audiofile = stream.download(output_path = "downloads/",
+                                                   filename = title +
+                                                              " - YouTube - " +
+                                                              video_id)
+                output_thumbnail = "downloads/square_thumbnail_" + video_id + \
+                                   ".jpg"
                 self.__make_squarethumb(yt.thumbnail_url, output_thumbnail)
                 with open(output_audiofile, "rb") as audio, \
                      open(output_thumbnail, "rb") as thumb:
                     await message.reply_audio(audio,
-                                              caption=video_url,
+                                              caption="[" + escaped_title +
+                                                      "](" + video_url + ")",
+                                              parse_mode="MarkdownV2",
                                               duration=length,
                                               performer=author,
                                               title=title,
                                               thumb=thumb)
                 os.remove(output_audiofile)
                 os.remove(output_thumbnail)
             else:
-                await message.reply("<code>No downloads for 10min+ audio " +
-                                    "or file size greater than 50M</code>",
-                                    parse_mode=ParseMode.HTML)
+                await message.reply("`No downloads for 10min\+ audio " +
```

Comment:
i doubt you need to escape "**+**", as with a dot it should work with no escaping

Reply:
same reason as the previous.